src/base/vec_storage.rs

```diff
@@ -31,6 +31,20 @@ pub struct VecStorage<T, R: Dim, C: Dim> {
     ncols: C,
 }
 
+/// The default implementation for the VecStorage struct.
+/// Requires R and C to implement the default trait.
+impl<T, R: Dim + std::default::Default, C: Dim + std::default::Default> Default
+    for VecStorage<T, R, C>
+{
+    fn default() -> Self {
+        Self {
+            data: Default::default(),
+            nrows: Default::default(),
+            ncols: Default::default(),
+        }
+    }
+}
+
```

Comment:
I don't think any dimensions *actually* implement `Default` at the moment :thinking: 

I think an alternative implementation could divide this into several cases:
- `R: DimName`, `C: DimName`. In this case we need to require `T: Default`
- `R: DimName`, `C = Dynamic`. In this case we can *avoid* requiring `T: Default` and create an "empty" matrix.
- `R = Dynamic`, `C: DimName`. Same as the previous.
- `R = Dynamic`, `C = Dynamic`. Same, we don't need `T: Default`.

This makes it possible to do
```rust
#[derive(Default)]
struct Wrapper<T> {
   vector: DVector<T>,
   matrix: DMatrix<T>,
}
```
without requiring `T: Default`. This is a common problem I'm frequently running into, at least.

Reply:
Note also that the proposed impl is unsound, because it always generates an empty vector regardless of whatever the dimensions are. This would lead to unsoundness whenever the dimensions are non-zero and any operation in nalgebra would try to index into the matrix.